src/hooks/useInterval.ts

```diff
@@ -1,27 +1,50 @@
-import { useEffect, useRef } from 'react'
+import { useEffect, useRef, useState, useCallback } from 'react'
+import isUndefinedOrNull from 'utils/isUndefinedOrNull'
+import useLastUpdated from './useLastUpdated'
 
-export default function useInterval(callback: () => void, delay: null | number, leading = true) {
-  const savedCallback = useRef<() => void>()
+export default function useInterval(callback: () => void, delay: undefined | null | number, leading = true) {
+  const [runImmediate, setRunImmediate] = useState(leading)
+  const [runAfter, setRunAfter] = useState(delay)
+  const savedCallback = useRef<() => void>(callback)
+  const [isReadyForUpdate, setIsReadyForUpdate] = useState(false)
+  const { lastUpdated, setLastUpdated: refresh } = useLastUpdated()
+
+  // Use props indirectly to not render set up timeout effect twice if both props changed.
+  useEffect(() => {
+    setRunImmediate(leading)
+    setRunAfter(delay)
+  }, [leading, delay])
+
+  const tick = useCallback(() => {
+    setIsReadyForUpdate(true)
+  }, [])
 
   // Remember the latest callback.
   useEffect(() => {
     savedCallback.current = callback
   }, [callback])
 
-  // Set up the interval.
   useEffect(() => {
-    function tick() {
-      const { current } = savedCallback
-      if (current) {
-        current()
-      }
+    if (isReadyForUpdate) {
+      savedCallback.current?.()
+      setIsReadyForUpdate(false)
+      setRunImmediate(false)
+      refresh()
     }
+  }, [isReadyForUpdate, refresh])
 
-    if (delay !== null) {
-      if (leading) tick()
-      const id = setInterval(tick, delay)
-      return () => clearInterval(id)
+  // Set up the timeout.
+  useEffect(() => {
+    if (!isUndefinedOrNull(runAfter)) {
+      if (runImmediate) tick()
+      else {
+        const id = setTimeout(tick, runAfter)
```

Comment:
To reduce calls by setting new timeout after calling a callback otherwise there will be more calls if user enters the exchange pages which triggers callback immediately and interval kicking in very close results unnecessary requests.